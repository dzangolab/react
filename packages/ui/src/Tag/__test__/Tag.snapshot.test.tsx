import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import { Tag } from "..";

describe("Tag", () => {
  test("should match snapshot with default props", () => {
    const { container } = render(<Tag />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with custom className", () => {
    const { container } = render(<Tag className="tag-item" />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with custom color set", () => {
    const { container } = render(<Tag color="green" />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with fullWidth set to true", () => {
    const { container } = render(<Tag fullWidth={true} />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with rounded set to true", () => {
    const { container } = render(<Tag rounded={true} />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with custom icon", () => {
    const { container } = render(<Tag icon="pi pi-times" />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with custom label", () => {
    const { container } = render(<Tag label="test tag" />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with custom style", () => {
    const { container } = render(<Tag style={{ color: "red" }} />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with custom renderContent", () => {
    const { container } = render(
      <Tag
        renderContent={() => {
          return (
            <>
              <i className="tag-item"></i>
            </>
          );
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with all props set", () => {
    const { container } = render(
      <Tag
        className="tag-item"
        color="green"
        fullWidth={true}
        icon="pi pi-times"
        label="Custom Label"
        rounded={true}
        style={{ color: "red" }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
