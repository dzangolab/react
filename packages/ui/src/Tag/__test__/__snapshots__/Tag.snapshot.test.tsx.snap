// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag > should match snapshot with all props set 1`] = `
<div>
  <span
    class="dz-tag tag-item rounded full-width"
    style="color: red; background-color: rgb(194, 236, 214);"
  >
    <i
      class="pi pi-times"
    />
    Custom Label
  </span>
</div>
`;

exports[`Tag > should match snapshot with custom className 1`] = `
<div>
  <span
    class="dz-tag tag-item"
    style="background-color: rgb(215, 194, 253);"
  />
</div>
`;

exports[`Tag > should match snapshot with custom color set 1`] = `
<div>
  <span
    class="dz-tag"
    style="background-color: rgb(194, 236, 214);"
  />
</div>
`;

exports[`Tag > should match snapshot with custom icon 1`] = `
<div>
  <span
    class="dz-tag"
    style="background-color: rgb(215, 194, 253);"
  >
    <i
      class="pi pi-times"
    />
  </span>
</div>
`;

exports[`Tag > should match snapshot with custom label 1`] = `
<div>
  <span
    class="dz-tag"
    style="background-color: rgb(215, 194, 253);"
  >
    test tag
  </span>
</div>
`;

exports[`Tag > should match snapshot with custom renderContent 1`] = `
<div>
  <span
    class="dz-tag"
    style="background-color: rgb(215, 194, 253);"
  >
    <i
      class="tag-item"
    />
  </span>
</div>
`;

exports[`Tag > should match snapshot with custom style 1`] = `
<div>
  <span
    class="dz-tag"
    style="color: red; background-color: rgb(215, 194, 253);"
  />
</div>
`;

exports[`Tag > should match snapshot with default props 1`] = `
<div>
  <span
    class="dz-tag"
    style="background-color: rgb(215, 194, 253);"
  />
</div>
`;

exports[`Tag > should match snapshot with fullWidth set to true 1`] = `
<div>
  <span
    class="dz-tag   full-width"
    style="background-color: rgb(215, 194, 253);"
  />
</div>
`;

exports[`Tag > should match snapshot with rounded set to true 1`] = `
<div>
  <span
    class="dz-tag  rounded"
    style="background-color: rgb(215, 194, 253);"
  />
</div>
`;
